proposals/function-pointers.md

```diff
@@ -36,15 +36,25 @@ of a `delegate*` will use `calli` where invocation of a `delegate` will use `cal
 Syntactically though invocation is identical for both constructs.
 
 The ECMA-335 definition of method pointers includes the calling convention as part of the type signature (section 7.1).
-The default calling convention will be `managed`. Alternate forms can be specified by adding the appropriate modifier
-after the `delegate*` syntax: `managed`, `cdecl`, `stdcall`, `thiscall`, or `unmanaged`. Example:
+The default calling convention will be `managed`. Unmanaged calling conventions can by specified by putting an `unmanaged`
+keyword afer the `delegate*` syntax, which will use the runtime platform default. Specific unmanaged conventions can then
+be specified in brackets to the `unmanaged` keyword by specifying any type starting with `CallConv` in the
+`System.Runtime.CompilerServices` namespace. These types must come from the program's core library, and the set of valid
+combinations is platform-dependent.
 
 ``` csharp
+//This method has a managed calling convention. This is the same as leaving the managed keyword off.
+delegate* managed<int, int>;
+
+// This method will be invoked using whatever the default unmanaged calling convention on the runtime
+// platform is. This is platform and architecture dependent and is determined by the CLR at runtime.
+delegate* unmanged<int, int>;
+
 // This method will be invoked using the cdecl calling convention
-delegate* cdecl<int, int>;
+delegate* unmanged[CallConvCdecl] <int, int>;
 
-// This method will be invoked using the stdcall calling convention
-delegate* stdcall<int, int>;
+// This method will be invoked using the stdcall calling convention, and suppresses GC transition
+delegate* unmanaged[CallConvStdCall, CallConvSuppressGCTransition] <int, int>;
```

Comment:
I take it we are not allowing `unmanaged[StdCall, SuppressGCTransition]` then? I don't remember a decision being made here, but I may have missed that bit 😄

Reply:
I went for simplicity of implementation, I'll add a consideration note.

Reply:
Thanks, it's mostly a convenience thing but I think it would help greatly with readability and with typing when you have hundreds of function pointers you are declaring/reviewing (such as DirectX which is all COM `ThisCall` + `StdCall`)...